src/Resque/Component/Core/Process.php

```diff
@@ -123,7 +123,7 @@ public function wait()
             );
         }
 
-        pcntl_waitpid($this->pid, $this->status);
+        $status = pcntl_waitpid($this->pid, $this->status, $options);
```

Comment:
I agree, I considered it, but again didnt want to change the ABI / design.

I too agree unless the api is made for chaining and lends itself to that style of design (i.e some unit testing api's, dom traversal) its best to avoid returning $this.